File: Abstract Factory/Golang/TeamPromoConnectionFactory.go
package main

type TeamPromoConnectionFactory struct {}

func (teamPromoConnectionFactory *TeamPromoConnectionFactory) createRPCConnection() IRPCConnection  {
	return &TeamPromoRPCConnection{}
}

func (teamPromoConnectionFactory *TeamPromoConnectionFactory) createHTTPConnection() IHTTPConnection  {
	return &TeamPromoHTTPConnection{}
}

File: Abstract Factory/Dart/TeamPromoConnectionFactory.dart
import 'IConnectionFactory.dart';
import 'IRPCConnection.dart';
import 'IHTTPConnection.dart';
import 'TeamPromoHTTPConnection.dart';
import 'TeamPromoRPCConnection.dart';

class TeamPromoConnectionFactory implements IConnectionFactory {
  @override
  IHTTPConnection createHTTPConnection() {
    return TeamPromoHTTPConnection();
  }

  @override
  IRPCConnection createRPCConnection() {
    return TeamPromoRPCConnection();
  }
}
